### PR TITLE
Jonasz kubsik dev

### PR DESCRIPTION
**Pull request by JonaszKubsik**

### CODE REVIEW

**Review by gorkowykamil — COMMENTED** [review]
Bardzo dobra i solidna praca. Kod bardzo czytelny i uporządkowany. Dobre przygotowanie danych, skalowanie i wybór metryk. Jedyne do czego mogę się przyczepić, to nie wiem dlaczego najpierw trenowałeś modele, a dopiero potem robiłeś kroswalidację w celu znalezienia optymalnych parametrów, zamiast zrobić to wszystko podczas kroswalidacji, ale nie jest to żaden problem. Poza tym dobre wykresy i wnioski. Wszystko dobrze przeanalizowane oraz wytłumaczone.

**Comment by a-janczak** [comment]
Coś chyba tylko poszło nie tak z DummyClassifierem, bo (przynajmniej u mnie się tak wyświetla) F1 wynosi zero. Ogólnie masz wszystko opisane i wytłumaczone, może tylko przy porównywaniu wyników KNN i drzewka zbioru treningowego i testowego fajnie by wyglądał jakiś wykresik zamiast samych liczb, ale to już bardziej kosmetyczna kwestia. A poza tym, to chyba tyle z moich przemyśleń :))

**Review by Dmytro-Lifyentsyev — COMMENTED** [review]
Nie jestem pełny czy badanie hiperparametrów jest dobrze zrobione, ponieważ chyba doszło do wycieku danych. Przyjmujesz decyzję o wybraniu hiperparametrów na podstawie zbioru testowego, więc dobierasz do konkretnych danych (traci objektywność). Lepiej byłoby zastosować klasę GridSearchCV, która dzieli zbiór treningowy na treningowy i walidacyjny. Powtarza to n razy i uśrednia wynik, w tym czasie zbiór testowy jest nietykalny. 

Oprócz tego zadanie wygląda na dobrze zrobione.

**Comment by Dominik-Galus** [comment]
Hej, rozwiązanie jest spoko, udało Ci się ominąć praktycznie wyciek danych, ale podam pare wskazówek, zaczne od plusów i potem przejde do minusów.

Plusy:
- poprawna standaryzacja KNN
- dobra diagnoza overfitu drzewa z różnicą o około 20% między treningowym a testowym
- dawno nie widziałem żeby ktoś na modelach sklearn'owych nie użył GridSearchCV tylko pętlami hehe

Minusy:
- Wybrałeś hiperparametry obserwując zachowanie na zbiorze TESTOWYM, metodologicznie robi się to poprzez walidacje krzyżową na zbiorze treningowym, a test zostawić na jednorazową, końcową ocene wybranego modelu - tak jak mówiłem na wykładzie
- Dummy z F1=0 robi wrażenie bezużyteczności baseline'u, ale to artefakt bo most_frequent nie przewiduje ani jednego pozytywnego, więc recall=0 i F1=0, rzeczywisty punkt odniesienia to accuracy, warto raportować obie metryki, żeby porównanie nie było mylące
- wykres hiperparametrów pokazują tylko test f1 bez linii treningowej, więc trudno ocenić czy model się przeucza, imho przydałaby się ta krzywa treningowa
- rzadko opisujesz dlaczego podejmujesz pewne decyzje, głównie opisujesz co widać
- masz pewną redundancje cech Sex, IsAdultMale i IsChild się częściowo pokrywa, nie jest to błąd ale można było to skomentować

Moje sugestie na teraz to hiperparametry przez walidację krzyżową np. GridSearchCV na X_train), a X_test użyj raz na końcu do raportu, w porównaniu z baseline'em pokaż też accuracy, dodaj krzywą treningową do wykresów hiperparametrów i dopisz analizę "dlaczego"
